resources/js/provider.js

```diff
@@ -1,242 +1,249 @@
-if(typeof EthereumProvider === "undefined"){
-var callbackId = 0;
-var callbacks = {};
-
-bridgeSend = function (data) {
-    ReactNativeWebView.postMessage(JSON.stringify(data));
-}
-
-function sendAPIrequest(permission, params) {
-    var messageId = callbackId++;
-    var params = params || {};
-
-    bridgeSend({
-        type: 'api-request',
-        permission: permission,
-        messageId: messageId,
-        params: params
-    });
-
-    return new Promise(function (resolve, reject) {
-        params['resolve'] = resolve;
-        params['reject'] = reject;
-        callbacks[messageId] = params;
-    });
-}
-
-function qrCodeResponse(data, callback){
-    var result = data.data;
-    var regex = new RegExp(callback.regex);
-    if (!result) {
-        if (callback.reject) {
-            callback.reject(new Error("Cancelled"));
-        }
+(function(){
```

Comment:
@flexsurfer a timely move! :)

I have a question though - why do we need to wrap in `(function() {...})()`. I understand that this is because we don't want the variables to pollute global namespace, correct?

Reply:
correct

Reply:
@siphiuel i also have a question, why does it work locally and doesn't in PR build ? :)

Reply:
ok , its because of this `msg (str "ReactNativeWebView.onMessage('"` but why does it work locally, have no idea

Reply:
Ahh, right, as you're now in function scope, you have to prepend `window.`.